flink-core/src/test/java/org/apache/flink/core/memory/HybridOnHeapMemorySegmentTest.java

```diff
@@ -77,4 +82,54 @@ public void testHybridHeapSegmentSpecifics() {
 		assertEquals(3, buf2.position());
 		assertEquals(7, buf2.limit());
 	}
+
+	@Test
+	public void testReadOnlyByteBufferPut() {
+		final byte[] buffer = new byte[100];
+		HybridMemorySegment seg = new HybridMemorySegment(buffer, null);
+
+		String content = "hello world";
+		ByteBuffer bb = ByteBuffer.allocate(20);
+		bb.put(content.getBytes());
+		bb.rewind();
+
+		int offset = 10;
+		int numBytes = 5;
+
+		ByteBuffer readOnlyBuf = bb.asReadOnlyBuffer();
+		assertFalse(readOnlyBuf.isDirect());
+		assertFalse(readOnlyBuf.hasArray());
+
+		seg.put(offset, readOnlyBuf, numBytes);
+
+		// verify the area before the written region.
+		for (int i = 0; i < offset; i++) {
+			assertEquals(0, buffer[i]);
+		}
+
+		// verify the region that is written.
+		assertEquals("hello", new String(buffer, offset, numBytes));
+
+		// verify the area after the written region.
+		for (int i = offset + numBytes; i < buffer.length; i++) {
+			assertEquals(0, buffer[i]);
+		}
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testReadOnlyByteBufferGet() {
```

Comment:
this test is basically already covered by `MemorySegmentTestBase.testHeapByteBufferGetReadOnly `

Reply:
Yes, I agree with you. Do you suggestion removing this test?

Reply:
Yes, I do not see any value in it.

Btw, just a side thought. We could keep normal handling of this weird case (non-direct and non-array-based target buffer). The problem is that to test it we would have to implement a testing buffer extending `ByteBuffer`. The testing buffer would have a lot of methods to implement because we cannot easily extend any of internal `ByteBuffer` implementations. I think the effort is not worth while spending time atm but we can still adopt it in future if needed.

Reply:
Thanks for your suggestion. I have removed this test case.